execute/wait.go

```diff
@@ -66,24 +66,28 @@ func waitForHTTPRequest(host string, cancel <-chan struct{}) {
 
 	err := errors.New("init")
 	for err != nil {
+		log.Printf("HTTP: HEAD %s\n", url.String())
 		req, reqErr := http.NewRequest("HEAD", url.String(), nil)
 		if reqErr != nil {
 			log.Printf("Warning: Failed to create request for URL '%s' -  skipping service '%s'",
 				url.String(), host)
 			return
 		}
+		req.Cancel = cancel
 
-		tr := &http.Transport{}
-		client := &http.Client{Transport: tr}
+		client := &http.Client{Transport: &http.Transport{}, Timeout: 500 * time.Millisecond}
 		c := make(chan error, 1)
 		go func() {
-			_, err := client.Do(req)
+			resp, err := client.Do(req)
+			if err == nil && resp.StatusCode != http.StatusOK {
+				err = errors.New(fmt.Sprintf("Expecting %v: got %v",
+					http.StatusOK, resp.Status))
```

Comment:
Use `fmt.Errorf` instead

    err = fmt.Errorf("expected HTTP status %v but got %v", ...)

Also, error strings usually start with lower case so that you can combine them with ':'. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings